### PR TITLE
RFC: Pluggable Repos

### PR DESCRIPTION
**Pull request by pauljamescleary**
Add the ability to configure repos to enable creation of repositories for different database backends.

### CODE REVIEW

File: text/0000-pluggable-repos.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  recordSet { // record set table properties go here, throughput, name }
+  recordSetChange  { // record set change properties go in here}
+}
+]
+```
+
+In following the proposed configuration, we would have the following...
+
+```scala
+trait DataStore {
```
Comment (rebstar6) [inline text/0000-pluggable-repos.md]: why optional? this should fail if were missing one...

File: text/0000-pluggable-repos.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+object DataStore {
+  
+  /* For each data store, load and initialize it.  Note: these can be done in parallel */
+  def load(config: Config): Seq[DataStore] = ...
+}
+
+...
+
+class Repos(
```
Comment (rebstar6) [inline text/0000-pluggable-repos.md]: hard to reason in this format, but the distinction between `Repos` and `DataStore` is unclear to me

File: text/0000-pluggable-repos.md (hunk 1 above)
Comment (rebstar6) [reply text/0000-pluggable-repos.md]: Datastore as in mysql vs dynamo? interesting. is there actual value in grouping all the mysqls into one and the dynamos into another? why not just have the repos and take out this layer?

Comment (pauljamescleary) [reply text/0000-pluggable-repos.md]: There are a few things we are looking for:

1. Specify data store configuration once for all "tables" / repositories
2. Allow each data store to initialize itself (run flyway migrations, create dynamodb tables)
3. Allow a mashup of repositories (like we do now).  Some repos are better in datastore x, others in datastore y

The configuration is the important point.  We have a `DataStore -- 1 to many --> Repository` in the model.

I think the requirements are what matters.  Could remove the implementation here.

**Pull request merged**